Widgets/UIHelpers.cs

```diff
@@ -9,14 +9,35 @@ public class UIHelpers
     {
         //--------------------------------------------------------------------------------------
         /// <summary>
-        /// GetValueFromText - Convert text from a vwml file into an value
+        /// GetValueFromText - Takes a type as a parameter and tries to convert it
+        /// into a string for GetBasicValueFromText
         /// </summary>
         //--------------------------------------------------------------------------------------
         public static object GetValueFromText(Type type, string valueText)
         {
-            if(type.IsArray) return ParseArray(type, valueText);
+            try
+            {
+                return GetBasicValueFromText(type.Name, valueText);
+            }
+            catch (Exception error)
+            {
+                if (type.IsArray) return ParseArray(type, valueText);
+                else if (type.Name.StartsWith("Tuple")) return ParseTuple(type, valueText);
+                else if (type.IsEnum) return Enum.Parse(type, valueText);
+                else if (type.IsClass) return Activator.CreateInstance(type, valueText);
+                else throw error;
+            }
+        }
+
 
-            switch (type.Name)
+        //--------------------------------------------------------------------------------------
+        /// <summary>
+        /// GetBasicValueFromText - Convert text from a vwml file into an value
+        /// </summary>
+        //--------------------------------------------------------------------------------------
+        public static object GetBasicValueFromText(string type, string valueText)
```

Comment:
This should not be public.  There should only be one API for getting a value from text.